pkg/bpf/prog_linux.go

```diff
@@ -122,6 +125,8 @@ func GetProgInfoByFD(fd int) (ProgInfo, error) {
 		duration = spanstat.Start()
 	}
 	ret, _, err := unix.Syscall(unix.SYS_BPF, BPF_OBJ_GET_INFO_BY_FD, uintptr(unsafe.Pointer(&attr)), unsafe.Sizeof(attr))
```

Comment:
Do we also need, runtime.KeepAlive(&attr) here?

Reply:
@jrfastab Good catch, `runtime.KeepAlive(&attrInfo)` below should have been `runtime.KeepAlive(&attr)` instead. However, neither public functions from `prog_linux.go` are being used, so maybe it makes sense to backport https://github.com/cilium/cilium/pull/10187 to the `v1.{5,6,7}` instead of fixing the dead code?

Reply:
I agree backporting dead-code removal makes sense here. I'll rework the backport series as well.